sklearn/ensemble/voting_classifier.py

```diff
@@ -13,12 +13,44 @@
 
 import numpy as np
 
+from .base import _partition_estimators
 from ..base import BaseEstimator
 from ..base import ClassifierMixin
 from ..base import TransformerMixin
 from ..base import clone
 from ..preprocessing import LabelEncoder
 from ..externals import six
+from ..externals.joblib import Parallel, delayed
+from ..utils.validation import has_fit_parameter
+
+
+def _parallel_fit_estimators(estimators, X, y, sample_weight):
+    """Private function used to fit a batch of estimators within a job."""
+    fitted_estimators = []
+    for estimator in estimators:
+        if sample_weight is not None and has_fit_parameter(estimator, "sample_weight"):
+            estimator.fit(X, y, sample_weight)
+        else:
+            estimator.fit(X, y)
+        fitted_estimators.append(estimator)
+    return fitted_estimators
+
+
+def _parallel_predict(estimators, X):
+    """Private function used to compute predictions within a job."""
+    return [estimator.predict(X) for estimator in estimators]
+
+
+def _parallel_predict_proba(estimators, X):
+    """Private function used to compute (proba-)predictions within a job."""
+    return [estimator.predict_proba(X) for estimator in estimators]
+
+
+def _flatten_list(list_):
```

Comment:
check out for example `bagging.py:367` where `list(itertools.chain.from_iterable(...))` is used instead of a special flatten function. Would be good if possible to keep the code consistent.